docs/DeveloperGuide.md

```diff
@@ -2,12 +2,343 @@
++ `dayOfRecurrence` (type `int`): stores the day of it own private attribute `date`, used for internal program checking
+usage
+
+The methods implemented in this class are a collection of getters and setters that allow other class types to access
+the information of the entry.
+
+The following are child classes of `EntryType`:
++ `Income`: Stores entries that the user labels as income
++ `Spending`: Stores entries that the user labels as spending
+
+#### Income class
+The `Income` class inherits from `EntryType` class. It is used to store relevant information for entries labelled as 
+income. This information was used by other classes to perform their component tasks.
+
+#### Spending class
+The `Spending` class inherits from `EntryType` class. It is used to store relevant information for entries labelled as
+spending. This information was used by other classes to perform their component tasks.
+
+### Storage Component
+#### Motivation behind the component:
++ Allows the user to save changes, so that they can resume where they left off.
++ Allows advanced users to edit files directly, enabling fast, manual adjustments.
+
+#### How the Storage Component works:
++ Variables and File Path:
+    + `incomes` → `./incomes.txt`
+    + `spendings` → `./spendings.txt`
+    + `password` → `./password.txt`
+
++ To save edited lists:
+    + It is done when users type `bye` or after keyboard interrupts (i.e.Ctrl-c), which signals the end of the program.
+    + The lists are saved to a user-editable format in their respective files.
+
++ To load saved lists:
+  + It is done upon program startup, when `Wiagi` is constructed.
+  + Within the `Wiagi` constructor, it will create a new instance of `Storage`, which will then load the data at the 
+  `incomes` and `spendings` file paths to an `IncomeList` and `SpendingList` respectively.
+  + `Wiagi` will then retrieve the lists in `Storage` to initialise its lists.
+  + Data corruption in the file triggers an exception, often due to user-editing errors.
+  + For missing files (e.g., new users), files are created and the initialised lists will be empty.
+
++ To load password:
+  + The hashed password will simply be loaded from the password file.
+  + For missing files (e.g., new users), users will be prompted to set a new password at the start of the
+  program. The entered password will then be hashed and stored in a newly created password file.
+
+#### Implementation:
+#### Storage class
+The `Storage` class is a class that stores `incomes`, `spendings` and `password`. 
+Upon instantiation, it will call `IncomeListStorage.load()`, `SpendingListStorage.load()` and `LoginStorage.load()`, 
+which will initialise the variables in `Storage` respectively.
+
+#### save method in `IncomeListStorage` `SpendingListStorage`
+<img src="./Diagrams/Storage/saveStorageSequenceDiagram.png" alt="saveStorageSequenceDiagram" width="600" height="400"/><br>
```

Comment:
You may want to consider omitting method calls to JavaAPI such as `FileWriter`, only if you feel that it is not of importance.